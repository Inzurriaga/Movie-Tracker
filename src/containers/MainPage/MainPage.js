import React, { Component } from 'react';
import { fetchMovies } from "../../api/index";
import { connect } from 'react-redux';
import { getMovies } from '../../actions';
import SlideShow from "../SlideShow/SlideShow";


class MainPage extends Component {
  // Genres
  // https://api.themoviedb.org/3/genre/movie/list?language=en-US&api_key=3033b069b2d0b32039c7d85cfe10cfd3

  // Discover
  // https:api.themoviedb.org/3/discover/movie?page=1&include_video=false&include_adult=false&sort_by=popularity.desc&language=en-US&api_key=3033b069b2d0b32039c7d85cfe10cfd3

  // Get Images
  // http://image.tmdb.org/t/p/original/{Extension}

  // Discover


  render() {
    const { movies } = this.props;
    return (
      <div className="MainPage">
        {
          movies.length ?
          <SlideShow /> :
          <div>im loading</div>
        }
      </div>
    );
  }
}



export const mapStateToProps = (state) => ({
  movies: state.movies,
})


export default connect(mapStateToProps)(MainPage)
