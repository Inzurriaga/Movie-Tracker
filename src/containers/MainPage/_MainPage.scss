.MainPage {
  display: flex;
  height: 92vh;
  padding: 2em;
  width: 100%;
  max-width: 2000px;
  margin: auto;
  box-sizing: border-box;

  .left-option {
    display: flex;
    flex-direction: column;
    width: 60%;

    .bottom-option {
      margin-top: 2em;
      display: flex;
      height: 25%;

      .theater-box {
        display: flex;
        text-decoration: none;
        align-items: center;
        justify-content: center;
        background: #231f20;
        width: 50%;


        font-size: 1.2em;
        font-weight: 100;
        letter-spacing: 0.04em;
        text-transform: uppercase;
        text-shadow: 2px 2px 5px black;
        font-family: $font-Abel;
        color: white;

        background-size: contain;
        background-position: top center;
        transition: all .5s ease;
        box-shadow: 0px 2px 10px 3px #00000017;
        cursor: pointer;

        border-radius: 5px;


        &:first-child {
          margin-right: 2em;
        }
      }
    }
  }

  .right-option {
    display: flex;
    flex-direction: column;
    margin-left: 2em;
    width: 40%;

  }
}
