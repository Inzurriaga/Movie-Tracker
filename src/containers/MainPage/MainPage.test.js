import React from "react"
import { shallow } from "enzyme"
import { MainPage, mapStateToProps } from "./MainPage";

const mockMovies = [{namne:"hello"},{name: "what"},{name: "yes"}]

describe("MainPage", () => {
    describe("MainPage", () => {
        let wrapper;

        beforeEach(() => {
            wrapper = shallow(
                <MainPage movies={mockMovies}/>
            )
        })

        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot()
        })

    })

    describe("mapStateToProps", () => {

        it("should return an object", () => {
            const mockData = {
                movies: [],
                genres: [],
                category: []
            }
            const expected = {
                movies: []
            }
            const mockprops = mapStateToProps(mockData)
            expect(mockprops).toEqual(expected)
        })
        
    })
})
