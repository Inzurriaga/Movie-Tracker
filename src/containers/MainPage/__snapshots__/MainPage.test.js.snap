// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage MainPage should match snap shot 1`] = `
<div
  className="MainPage"
>
  <div
    className="left-option"
  >
    <Connect(SlideShow) />
    <div
      className="bottom-option"
    >
      <Link
        className="theater-box"
        style={
          Object {
            "backgroundImage": "url(theather.jpg)",
          }
        }
        to="/InTheater"
      >
        Movies in theater
      </Link>
      <Link
        className="theater-box"
        style={
          Object {
            "backgroundImage": "url(comingSoon.jpg)",
          }
        }
        to="/upcoming"
      >
        Coming out Soon
      </Link>
    </div>
  </div>
  <div
    className="right-option"
  >
    <Connect(GenreBox) />
  </div>
</div>
`;
